sympy/core/numbers.py

```diff
@@ -4279,7 +4279,15 @@ def all_close(expr1, expr2, rtol=1e-5, atol=1e-8):
     """
     NUM_TYPES = (Rational, Float)
 
-    def _all_close(expr1, expr2, rtol, atol):
+    def _all_close(obj1, obj2):
```

Comment:
regarding `_all_close_num` see https://stackoverflow.com/a/5595523/1089161 or https://stackoverflow.com/a/33024979/1089161

Reply:
> Perhaps this routine could be modified as following to allow `x` to match `1.0*x`:

I don't think that those should match. The goal here is to match structurally equivalent expressions where numeric components are possibly only approximately equal and do not necessarily have the same types like float vs rational.

I like the idea of using `as_coefficients_dict` but I don't think it works for Mul.

Maybe we could merge this as a fix for Groebner and then think about improving `all_close` in a separate PR.

Reply:
> ```python
> # XXX should it be rtol*max(abs(num1),abs(num2)))?
> ```

I don't have strong feelings about this but this is intentionally asymmetric because the intended use for this function is asymmetric as noted in the docstring:

>  Relative tolerance is measured with respect to expr2 so when used in testing expr2 should be the expected correct answer.